sdk/eventhub/azure-eventhub/samples/async_samples/send_async.py

```diff
@@ -16,41 +16,75 @@
 import os
 
 from azure.eventhub.aio import EventHubProducerClient
-from azure.eventhub import EventData, EventDataBatch
+from azure.eventhub import EventData
 
-EVENT_HUB_CONNECTION_STR = os.environ['EVENT_HUB_CONN_STR']
+CONNECTION_STR = os.environ['EVENT_HUB_CONN_STR']
 EVENTHUB_NAME = os.environ['EVENT_HUB_NAME']
 
 
-async def run(producer):
+async def send_event_data_batch(producer):
+    # Without specifying partition_id or partition_key
+    # The events will be distributed to available partitions via round-robin.
+    event_data_batch = await producer.create_batch()
+    event_data_batch.add(EventData('Single message'))
+    await producer.send_batch(event_data_batch)
 
-    async with producer:
 
-        # Without specifying partition_id or partition_key
-        # The events will be distributed to available partitions via round-robin.
-        event_data_batch = await producer.create_batch(max_size_in_bytes=10000)
+async def send_event_data_batch_with_limited_size(producer):
+    # Without specifying partition_id or partition_key
+    # The events will be distributed to available partitions via round-robin.
+    event_data_batch_with_limited_size = await producer.create_batch(max_size_in_bytes=1000)
+
+    while True:
+        try:
+            event_data_batch_with_limited_size.add(EventData('Message inside EventBatchData'))
+        except ValueError:
+            # EventDataBatch object reaches max_size.
+            # New EventDataBatch object can be created here to send more data
+            break
+
+    await producer.send_batch(event_data_batch_with_limited_size)
+
+
+async def send_event_data_batch_with_partition_key(producer):
+    # Specifying partition_key
+    event_data_batch_with_partition_key = await producer.create_batch(partition_key='pkey')
+    event_data_batch_with_partition_key.add(EventData('Message will be sent to target-key partition'))
```

Comment:
"Message will be sent to a partition determined by the partition key"

Reply:
good suggestion, updated

Reply:
nice words, updated